packages/twenty-server/src/workspace/workspace-sync-metadata/reflective-metadata.factory.ts

```diff
@@ -28,58 +45,80 @@ export class MetadataParser {
       (field) => !isGatedAndNotEnabled(field, workspaceFeatureFlagsMap),
     );
 
+    if (objectMetadata?.dataSourceSchema) {
+      const dataSource = await this.dataSourceRepository.findOne({
+        where: {
+          schema: objectMetadata.dataSourceSchema,
+        },
+      });
+
+      if (!dataSource) {
+        throw new InternalServerErrorException(
+          'Data source not found for object metadata',
+        );
+      }
+
+      dataSourceId = dataSource.id;
+
+      delete objectMetadata.dataSourceSchema;
+    }
+
     return {
       ...objectMetadata,
       workspaceId,
       dataSourceId,
-      fields: fields.map((field: FieldMetadataEntity) => ({
+      fields: fields.map((field) => ({
         ...field,
         workspaceId,
         isSystem: objectMetadata.isSystem || field.isSystem,
-        defaultValue: field.defaultValue || null,
-        options: field.options || null,
```

Comment:
can we keep options? I think we would like to add enum very soon for our standard objects 🙂

Reply:
I think it's better to add them again when we support the feature

Reply:
Ah yes makes sense 👍